### PR TITLE
Fix installer issue if SQL field has no precision

### PR DESCRIPTION
**Pull request by aschempp**
fixes https://github.com/isotope/core/pull/2351#issuecomment-1378849855

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Database/Installer.php [hunk 1]
```diff
@@ -369,7 +369,7 @@ public function getFromDb()
 					// Field type
 					if ($field['length'])
 					{
-						$field['type'] .= '(' . $field['length'] . ($field['precision'] ? ',' . $field['precision'] : '') . ')';
+						$field['type'] .= '(' . $field['length'] . (isset($field['precision']) ? ',' . $field['precision'] : '') . ')';
```
Comment (fritzmg) [inline core-bundle/src/Resources/contao/library/Contao/Database/Installer.php:372]:
```suggestion
						$field['type'] .= '(' . $field['length'] . (!empty($field['precision']) ? ',' . $field['precision'] : '') . ')';
```

Might be more safe?

Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Database/Installer.php:372]: I thought about this as well, but 
 1. we use `isset` for other such cases
 2. `empty` would return false for `0`. Even though that's probably an invalid value, we don't really want to validate the value itself but if something is set we want to use it?

**Review by fritzmg — DISMISSED** [review]

**Comment by ameotoko** [comment]
Doesn't help: https://github.com/isotope/core/pull/2351#issuecomment-1385179734

File: core-bundle/src/Resources/contao/library/Contao/Database/Installer.php [hunk 2]
```diff
@@ -369,7 +369,7 @@ public function getFromDb()
 					// Field type
 					if ($field['length'])
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/Database/Installer.php]:
```suggestion
					if ($field['length'] ?? null)
```

**Comment by ameotoko** [comment]
There's a lot of things to be fixed here, and they emerge one by one. Perhaps I should have been even more specific in https://github.com/isotope/core/issues/2343 from the very start. When I said `After some digging` in https://github.com/isotope/core/issues/2343#issuecomment-1209367458 – I meant that I already tried exactly the fixes proposed in this PR. There will be a lot of them!

When you suddenly face uncomparable amount of code you need to fix, it might be a sign that you're fixing it in the wrong place. So I went one level up. I am still convinced, that what I said in https://github.com/isotope/core/issues/2343#issuecomment-1209367458 makes sense:

> the actual failing method `Contao\Database\Installer->getFromDb()` is only used by Contao in `Contao\Database\Installer->generateSqlForm()`, which is deprecated since `4.0` and currently not used by Contao itself in any branch.

**Review by leofeyer — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Database/Installer.php [hunk 3]
```diff
@@ -367,9 +367,9 @@ public function getFromDb()
 					unset($field['index'], $field['origtype']);
 
 					// Field type
-					if ($field['length'])
+					if ($field['length'] ?? null)
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Database/Installer.php]:
```suggestion
					if (isset($field['length']))
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**